### ISSUE TITLE
Notifications: Provide notifications for Device/Instance crashes

### ISSUE THREAD

**Issue by joepavitt** [FlowFuse/flowfuse#4295]

**Comment by Steve-Mcl** [comment]
Joe, since the notification API seems to only support a single recipient who should receive said notifications?

**Comment by joepavitt** [comment]
Send multiple notifications? Find members of team, loop over, send notification to each?

**Comment by Steve-Mcl** [comment]
@joepavitt 

> Send multiple notifications? Find members of team, loop over, send notification to each?

members? or just owners? 

would that include admins of the platform too?

What if a team has 100+ members?

Also, not certain if/how/when we would clear this (if at all)?

**Comment by joepavitt** [comment]
@hardillb who receives an email when instances go down, members + owner, or just owners?

No need for platform admins to be made aware

**Comment by joepavitt** [comment]
> Also, not certain if/how/when we would clear this (if at all)?

If the user clicks the notification, they'll be navigated to the instance overview, at which point, notification is marked as read

**Comment by hardillb** [comment]
It's configurable

![image](https://github.com/user-attachments/assets/9973bc1f-0fa3-43de-a93a-c9cb8f78ae75)

**Comment by Steve-Mcl** [comment]
> > Also, not certain if/how/when we would clear this (if at all)?
> 
> If the user clicks the notification, they'll be navigated to the instance overview, at which point, notification is marked as read

Yeah, i know, but what if the application is restarted successfully (problem is resolved) - do we leave the notification in the other 99 users inbox?

**Comment by Steve-Mcl** [comment]
> It's configurable
> 
> ![image](https://github.com/user-attachments/assets/9973bc1f-0fa3-43de-a93a-c9cb8f78ae75)

That is specific to emails not notifications @hardillb (not is scope as far as I can see)

**Comment by hardillb** [comment]
@Steve-Mcl it's the answer to Joe's question

**Comment by joepavitt** [comment]
Can we read from that same setting to drive the in-app notification too?

**Comment by Steve-Mcl** [comment]
> Can we read from that same setting to drive the in-app notification too?

Yes, we can.

**Comment by Steve-Mcl** [comment]
> > > Also, not certain if/how/when we would clear this (if at all)?
> > 
> > 
> > If the user clicks the notification, they'll be navigated to the instance overview, at which point, notification is marked as read
> 
> Yeah, i know, but what if the application is restarted successfully (problem is resolved) - do we leave the notification in the other 99 users inbox?

@joepavitt ^

**Comment by joepavitt** [comment]
yes - we do

**Comment by Steve-Mcl** [comment]
> > Can we read from that same setting to drive the in-app notification too?
> 
> Yes, we can.

Bah - the email options are EE only. 

### Possible directions that spring to mind are (listed most difficult to easiest)

1. We modify the email preferences so that it is dual purpose (emails and notification) (hide email section for CE)
    *  Will require the `settings` data to be restructured due to current format 
    ```jsonc
    /* current settings object ... */
    { 
        "emailAlerts": { "crash": true,  "safe": true, "recipients": "both"  } 
    }
    /* would need to be transformed for all instances to ... */
    { 
        "alerts": {
            "email": { "crash": true, "safe": true },
            "notification": { "crash": true, "safe": true },
            "recipients": "both"
        }
    }
    ```
3. We duplicate the email preferences as "notifications" 
   * This is cleaner in implementation due to separation of concerns
   * Settings can be stored without affecting existing structure
    ```jsonc
    /* existing settings object - unchanged */
    {
        "emailAlerts": { "crash": true,  "safe": true, "recipients": "both"
    }
    /* add new setting notificationAlerts */
    {
        "notificationAlerts": { "crash": true,  "safe": true, "recipients": "both" } 
    }
    ```
4. We just send the crashed notification to all members + owners (no options)
5. We just send the crashed notification to owners (no options)
6. We dont send crash notification on CE platforms (i.e. becomes an EE feature)


### Mockups

![image](https://github.com/user-attachments/assets/903ba6c1-3bd1-45c5-aa08-ee68ec9d5c00)

![image](https://github.com/user-attachments/assets/5bd6ee75-8e6b-4dfd-8c0d-8709b0507a6b)

<br>

@joepavitt I couldnt find in the owner/parent issues if notifications are EE only but since they are NOT implemented under the EE file structure I have assumed this feature is for all?

**Comment by joepavitt** [comment]
Correct this is not an EE-Only feature - first iteration, send to all.

We can offer configurations for this at a later date.
